cds/modules/ffmpeg/ffmpeg.py

```diff
@@ -59,14 +66,47 @@ def ff_probe_all(input_filename):
     * *-show_format -print_format json* output in JSON format
     * *-show_streams -select_streams v:0* show information for video streams
     """
-    return check_output([
+    metadata = check_output([
         'ffprobe', '-v', 'error',
         '-show_format', '-print_format', 'json',
         '-show_streams', '-select_streams', 'v:0',
         '{}'.format(input_filename)
     ]).decode('utf-8')
 
+    # Patch invalid aspect ratio
+    info = json.loads(metadata)
+    sinfo = info['streams'][0]
+    key = 'display_aspect_ratio'
+    if sinfo[key] == '0:1':
+        sinfo[key] = calculate_aspect_ratio(sinfo['width'], sinfo['height'])
+    return json.dumps(info)
+
+
+#
+# Aspect Ratio
+#
+def valid_aspect_ratios():
+    return [(16, 9), (4, 3), (3, 2), (20, 9), (256, 135), (64, 35), (2, 1)]
 
+
+def probe_aspect_ratio(input_filename):
```

Comment:
I would factor out https://github.com/CERNDocumentServer/cds/pull/372/files#diff-2fee8efc94536f46efb65a2abb918206R76 into a function instead, it looks more natural